crates/store/re_protos/proto/rerun/v0/redap.proto

```diff
@@ -0,0 +1,275 @@
+    rpc UpdateResourceProperties(UpdateResourcePropertiesRequest)
+        returns (UpdateResourcePropertiesResponse) {}
+
+    // Retrieve the table of properties for all resources stored in the collection.
+    //
+    // Possibly filtered / column projected.
+    //
+    // The schema of the returned Dataframe should match the results of `PropertyTableSchema`.
+    rpc GetResourceProperties(GetResourcePropertiesRequest)
+        returns (stream rerun.common.v0.DataframePart) {}
+
+    // Register one or more resources with catalog entry
+    //
+    // This API is only supported for some catalog entries.
+    rpc RegisterResources(RegisterResourcesRequest) returns (rerun.common.v0.DataframePart) {}
+
+    // Remove a resource from the a manually-managed catalog entry
+    //
+    // This API is only supported for some catalog entries.
+    rpc UnregisterResource(UnregisterResourcesRequest) returns (UnregisterResourceResponse) {}
+
+    // Remove all resources from a manually-managed catalog entry
+    //
+    // This API is only supported for some catalog entries.
+    rpc UnregisterAllResources(UnregisterAllResourcesRequest)
+        returns (UnregisterAllResourcesResponse) {}
+
+    // Create an index for a specific column in the catalog entry
+    rpc CreateIndex(CreateIndexRequest) returns (CreateIndexResponse) {}
+
+    // Search over an existing index  in the catalog entry.
+    //
+    // The response to `SearchIndex` a RecordBatch with 3 columns:
+    // - 'resource_id' column with the id of the resource
+    // - timepoint column with the values representing the points in time
+    // where index query matches. What time points are matched depends on the type of
+    // index that is queried. For example for vector search it might be timepoints where
+    // top-K matches are found within *each* resource in the indexed entry. For inverted index
+    // it might be timepoints where the query string is found in the indexed column
+    // - 'data' column with the data that is returned for the matched timepoints
+    rpc SearchIndex(SearchIndexRequest) returns (stream rerun.common.v0.DataframePart) {}
+}
+
+// The Resource service is for accessing and querying the contents of individual resources.
+//
+// Each request includes a `resource_id` field, which is used to uniquely identify the resource within the catalog.
+service Resource {
+    // Get the schema of a resource
+    rpc GetResourceSchema(GetResourceSchemaRequest) returns (GetResourceSchemaResponse) {}
+
+    // Get information about the chunks that make up the resource.
+    //
+    // This includes the ability to filter the chunks by time-range or entity-path.
+    //
+    // ChunkInfo contains metadata about the chunk but not the data itself.
+    rpc GetChunkInfos(GetChunkInfosRequest) returns (stream ChunkInfo) {}
+
+    // Retrieve specific chunks of data from the resource.
+    rpc GetChunks(stream GetChunkRequest) returns (stream rerun.common.v0.RerunChunk) {}
```

Comment:
```suggestion
    rpc GetChunks(GetChunkRequest) returns (stream rerun.common.v0.RerunChunk) {}
```

`stream` in request position doesn't work on web. `GetChunksRequest` would contain `chunk_ids` field with a list of chunk IDs to retrieve. This is how we make it work for web

Reply:
We also need a way to stream chunks to the resources to replace `WriteMessages`.

Reply:
We should add some sort of lint/CI job to catch this common error